app/main/RTMediaUploadTerms.php

```diff
@@ -0,0 +1,142 @@
+			wp_localize_script( 'rtmedia-upload-terms-main', 'rtmedia_upload_terms_check_terms_message', esc_js( apply_filters( 'rtmedia_upload_terms_check_terms_message', $general_upload_terms_error_message ) ) );
+			wp_localize_script( 'rtmedia-upload-terms-main', 'rtmedia_upload_terms_check_terms_default_message', esc_js( apply_filters( 'rtmedia_upload_terms_check_terms_default_message', __( 'Please check Terms of Service.', 'buddypress-media' ) ) ) );
+		}
+
+		/**
+		 * Render terms and service checkbox in media upload tab.
+		 *
+		 * @global object $rtmedia
+		 * @param  string $content Content after I agree checkbox.
+		 * @return string
+		 */
+		public function show_terms_and_service_checkbox( $content ) {
+			global $rtmedia;
+
+			$options       = $rtmedia->options;
+			$terms_content = '';
+
+			if ( ( ! empty( $options['general_enable_upload_terms'] ) && '0' !== $options['general_enable_upload_terms'] ) && ( ! empty( $options['general_upload_terms_page_link'] ) && '' !== $options['general_upload_terms_page_link'] ) ) {
+				$terms_content = $this->terms_and_service_checkbox_html( $options );
+			}
+
+			return $content . $terms_content;
+		}
+
+		/**
+		 * Adding checkbox on user activity screen.
+		 *
+		 * @param  string $content Content for adding checkbox on user activity screen.
+		 * @return string
+		 */
+		public function show_terms_and_service_checkbox_activity( $content ) {
+			global $rtmedia;
+
+			$options       = $rtmedia->options;
+			$terms_content = '';
+
+			if ( ( ! empty( $options['activity_enable_upload_terms'] ) && '0' !== $options['activity_enable_upload_terms'] ) && ( ! empty( $options['general_upload_terms_page_link'] ) && '' !== $options['general_upload_terms_page_link'] ) ) {
+				$terms_content = $this->terms_and_service_checkbox_html( $options );
+			}
+
+			return $content . $terms_content;
+		}
+
+		/**
+		 * Checkbox of agree terms and condition at front-end.
+		 *
+		 * @param array $options Options set from rtMedia settings.
+		 *
+		 * @return string
+		 */
+		public function terms_and_service_checkbox_html( $options ) {
+			$general_upload_terms_page_link = $options['general_upload_terms_page_link'];
+			$general_upload_terms_message   = $options['general_upload_terms_message'];
+			$content                        = '';
+			ob_start();
+			?>
+			<div class="rtmedia-upload-terms">
+				<input type="checkbox" name="rtmedia_upload_terms_conditions" id="rtmedia_upload_terms_conditions" />
+				<label for="rtmedia_upload_terms_conditions">
+				<?php esc_html_e( apply_filters( 'rtmedia_upload_terms_service_agree_label', __( 'I agree to', 'buddypress-media' ) ) ); ?>
+				&nbsp;
+				<a href='<?php esc_url_e( $general_upload_terms_page_link ); ?>' target='_blank'>
+				<?php esc_html_e( apply_filters( 'rtmedia_upload_terms_service_link_label', $general_upload_terms_message ), 'buddypress-media' ); ?>
```

Comment:
@malavasita Check `esc_html_e` for usage. Parameters should be string literal and not a variable.